components/product-media-gallery/assets/product-media-gallery.js

```diff
@@ -22,10 +22,14 @@ export class ProductMediaGallery extends HTMLElement {
       return
     }
 
-    window.scroll({
-      top: mediaContainer.offsetTop,
-      behavior: 'smooth'
-    })
+    if (window.matchMedia('screen and (max-width: 769px)').matches) {
+      this.scrollTo({ left: mediaContainer.offsetLeft - (this.offsetWidth - mediaContainer.offsetWidth) / 2, behavior: 'smooth' })
+    } else {
+      window.scroll({
+        top: mediaContainer.offsetTop,
+        behavior: 'smooth'
+      })
+    }
```

Comment:
Alternative would be to isolate logic, your call :)
```suggestion
    const isMobile = window.matchMedia('screen and (max-width: 768px)').matches
    const scrollOptions = isMobile
      ? { left: mediaContainer.offsetLeft - (this.offsetWidth - mediaContainer.offsetWidth) / 2, behavior: 'smooth' }
      : { top: mediaContainer.offsetTop, behavior: 'smooth' }
      
    isMobile ? this.scrollTo(scrollOptions) : window.scroll(scrollOptions)
```

Reply:
Also note I updated the value based on https://github.com/archetype-themes/reference-components/blob/main/styles/layout/breakpoints.css#L1.

Reply:
Nice write up, logic is clear 👍